### PR TITLE
Add org/com_id, migrating, migrated_at

### PR DESCRIPTION
**Pull request by svenfuchs**

### CODE REVIEW

File: db/main/migrate/20180222000000_add_com_id_and_org_id.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class AddComIdAndOrgId < ActiveRecord::Migration
+  def change
+    add_column :branches, :org_id, :integer
```
Comment (aakritigupta) [inline db/main/migrate/20180222000000_add_com_id_and_org_id.rb]: You might want to add `NOT_NULL` to these as well.

File: db/main/migrate/20180222000002_add_migrating.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class AddMigrating < ActiveRecord::Migration
+  def change
+    add_column :repositories, :migrating, :boolean
+    add_column :repositories, :migrated_at, :datetime
+
+    add_column :organizations, :migrating, :boolean
+    add_column :users, :migrating, :boolean
```
Comment (aakritigupta) [inline db/main/migrate/20180222000002_add_migrating.rb]: SO we don't need `migrated_at` for `users` at all? Are we only going to migrate data per repo?

Comment (aakritigupta) [reply db/main/migrate/20180222000002_add_migrating.rb]: OK, I see it is added in a later migration in this PR.

**Comment by aakritigupta** [comment]
@svenfuchs this generally looks fine to me. Just that, you could bundle up the adding of `com_id` and `org_id` migrations into one migration, instead of making several different ones. Just easier to rollback/test if needed.

**Comment by aakritigupta** [comment]
One question though, how do you plan to populate the `{org, com}_id` fields? Somewhere when migrating the data?

File: db/main/migrate/20180222000000_add_com_id_and_org_id.rb (hunk 1 above)
Comment (svenfuchs) [reply db/main/migrate/20180222000000_add_com_id_and_org_id.rb]: @aakritigupta no, i don't think so. they're going to be `null` quite often?

Comment (igorwwwwwwwwwwwwwwwwwwww) [reply db/main/migrate/20180222000000_add_com_id_and_org_id.rb]: please make sure all new columns are nullable, as adding a `not null` column to a large table in production is not a safe operation. see also: https://www.braintreepayments.com/blog/safe-operations-for-high-volume-postgresql/

**Comment by svenfuchs** [comment]
Rebasing and re-arranging these was quite ... a task :)

@aakritigupta I've made it so that we now have 4 separate migration files, and they come in two commits :)

the org/com_id fields are being populated during the import process, yes

File: db/main/migrate/20180222000000_add_com_id_and_org_id.rb (hunk 1 above)
Comment (aakritigupta) [reply db/main/migrate/20180222000000_add_com_id_and_org_id.rb]: That was a good read, thanks for pointing to it @igorwwwwwwwwwwwwwwwwwwww 

@svenfuchs the column won't be `null` often and it is not worth adding the default value in a migration, so, let's keep it as is :)

**Comment by svenfuchs** [comment]
all shipped. one index is still being created on com, but i expect that to pass in ~30min. gonna merge this.

**Pull request merged**